modules/proxyconnection/ProxyConnectionPC.js

```diff
@@ -0,0 +1,392 @@
+            p2p: {
+                iceServers: [
+                    {
+                        urls: 'stun:stun.l.google.com:19302'
+                    },
+                    {
+                        urls: 'stun:stun1.l.google.com:19302'
+                    },
+                    {
+                        urls: 'stun:stun2.l.google.com:19302'
+                    }
+                ]
+            },
+            jvb: { iceServers: [] }
+        };
+
+        /**
+         * {@code JingleSessionPC} expects an instance of
+         * {@code JitsiConference}, which has an event emitter that is used
+         * to signal various connection updates that the local client should
+         * act upon. The conference instance is not a dependency of a proxy
+         * connection, but the emitted events can be relevant to the proxy
+         * connection so the event emitter is stubbed.
+         *
+         * @param {string} event - The constant for the event type.
+         * @type {Function}
+         * @returns {void}
+         */
+        const emitter = event => {
+            switch (event) {
+            case XMPPEvents.CONNECTION_ICE_FAILED:
+            case XMPPEvents.CONNECTION_FAILED:
+                this._onError(ACTIONS.CONNECTION_ERROR, event);
+                break;
+            }
+        };
+
+        /**
+         * {@code JingleSessionPC} expects an instance of
+         * {@code JitsiConference} to be passed in. {@code ProxyConnectionPC}
+         * is instantiated outside of the {@code JitsiConference}, so it must be
+         * stubbed to prevent errors.
+         *
+         * @type {Object}
+         */
+        const roomStub = {
+            addPresenceListener: () => { /** no-op */ },
+            connectionTimes: [],
+            eventEmitter: { emit: emitter },
+            getMediaPresenceInfo: () => {
+                // Errors occur if this function does not return an object
+
+                return {};
+            },
+            removePresenceListener: () => { /** no-op */ }
+        };
+
+        /**
+         * Create an instance of {@code RTC} as it is required for peer
+         * connection creation by {@code JingleSessionPC}. An existing instance
+         * of {@code RTC} from elsewhere should not be re-used because it is
+         * a stateful grouping of utilities.
+         */
+        const rtc = new RTC(this, {});
+
+        /**
+         * Add the remote track listener here as {@code JingleSessionPC} has
+         * {@code TraceablePeerConnection} which uses {@code RTC}'s event
+         * emitter.
+         */
+        rtc.addListener(
+            RTCEvents.REMOTE_TRACK_ADDED,
+            this._onRemoteStream
+        );
+
+        const peerConnection = new JingleSessionPC(
+            undefined, // sid
+            undefined, // localJid
+            this._options.peerJid, // remoteJid
+            connectionStub, // connection
+            {
+                offerToReceiveAudio: false,
```

Comment:
I know this is tailored to our specific needs, but how about adding `receiveAudio` as an option as well ? I was thinking it could be more usable for something else this way.